rs/index/src/ivf/files/invalidated_ids.rs

```diff
@@ -131,6 +141,57 @@ impl InvalidatedIdsStorage {
 
         Ok(())
     }
+
+    pub fn iter(&mut self) -> InvalidatedIdsIterator {
+        InvalidatedIdsIterator {
+            files: (0..self.files.len())
+                .filter_map(|i| {
+                    OpenOptions::new()
+                        .read(true)
+                        .open(format!("{}/invalidated_ids.bin.{}", self.base_directory, i))
+                        .ok()
+                })
+                .collect(),
+            current_file_idx: 0,
+            current_offset: 0,
+        }
+    }
+}
+
+impl Iterator for InvalidatedIdsIterator {
+    type Item = InvalidatedId;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        const INVALIDATED_ID_SIZE: usize = size_of::<u128>() + size_of::<u128>();
+
+        while self.current_file_idx < self.files.len() {
+            let file = &mut self.files[self.current_file_idx];
+
+            // Get the current file size
+            let file_size = file.metadata().unwrap().len() as usize;
+
+            if self.current_offset >= file_size {
+                // Move to the next file if we've reached the end of the current file
+                self.current_file_idx += 1;
+                self.current_offset = 0;
+                continue;
+            }
+
```

Comment:
might be worth it to check that `self.current_offset < file_size - INVALIDATED_ID_SIZE`